plugins/modules/ec2_instance.py

```diff
@@ -1289,6 +1299,15 @@ def build_top_level_options(params):
         spec['InstanceInitiatedShutdownBehavior'] = params.get('instance_initiated_shutdown_behavior')
     if params.get('termination_protection') is not None:
         spec['DisableApiTermination'] = params.get('termination_protection')
+    if params.get('hibernation_options') is not None and params.get('volumes'):
```

Comment:
`hibernation_options` is a bool and default on `False`. So you can just do:

`if params.get('hibernation_options') and params.get('volumes'):`